cobc/cobc.c

```diff
@@ -3855,6 +3862,32 @@ process_command_line (const int argc, char **argv)
 			}
 			break;
 
+		case CB_FLAG_GETOPT_SOURCE_ENCODE:
+			/* -fsource-encode=encoding*/
+			const char* valid_encodings[] = {
+				"UTF-8",
+				"ASCII",
+				"ISO-8859-1",
+				"ISO-8859-15"
+			};
```

Comment:
won't iconv return an error if it is executed with a "bad" encoding identifier? Then we need that only for the !`HAVE_ICONV` case an d can otherwise just pass it to iconv as-is.

Reply:
iconv will return an error, but what if the input source encoding is correct but not supported like UTF-32 or UTF-16?